dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -191,8 +191,14 @@ protected void run(@Nonnull Connector connector, @Nonnull ConnectorArguments arg
         // We had a customer request to base it on the database, but that isn't well-defined,
         // as there may be 0 or N databases in a single file.
         File outputFile = arguments.getOutputFile();
+        String outputDirectory = arguments.getOutputDirectory();
+
+        if ((outputDirectory == null) == (outputFile == null)) {
```

Comment:
The condition `((outputDirectory == null) == (outputFile == null))` is difficult to understand. Consider extracting the subexpressions to appropriately named local variables and construct the condition using such variables to make the whole condition more readable.